src/pages/App.tsx

```diff
@@ -185,6 +185,18 @@ export default function App() {
     [account]
   )
 
+  // redirect address to landing pages until implemented
+  const shouldRedirectToAppInstall = pathname?.startsWith('/address/')
+  useLayoutEffect(() => {
+    if (shouldRedirectToAppInstall) {
+      window.location.href = getDownloadAppLink()
+    }
+  }, [shouldRedirectToAppInstall])
+
+  if (shouldRedirectToAppInstall) {
```

Comment:
can this be inside of the first `if`?

Reply:
the first effect redirects and the second if just avoids rendering stuff, not sure which you mean specifically. i was worried about redirecting directly in the render as i'm not sure if thats fully synchronous and it feels more like a side-effect to me, but it's possible we could just do this entirely in the single render / if?

Reply:
the effect could go in the if as well but i guess technically weird looking as it breaks rules of hooks, though in this case i don't think there's any issue i just feel like worth avoiding the weirdness

Reply:
this is what I was imagining to reduce redundancy
```
useLayoutEffect(() => {
    if (shouldRedirectToAppInstall) {
      window.location.href = getDownloadAppLink()
      return null
    }
  }, [shouldRedirectToAppInstall]) 
  ```
  And just getting rid of the second if statement

Reply:
the return null is in the actual component render though, the intention is to avoid displaying anything / doing extra work while the browser redirects